### ISSUE TITLE
is there a way to tell it to generate new circuit (new IP)?

### ISSUE THREAD

**Issue by airdogvan** [dperson/torproxy#49]
I'm running with the simple docker run -it -p 8118:8118 -p 9050:9050 -d dperson/torproxy  command and using it at port 9050 for browser proxy.
But I noticed each time I start it it seems to run the same circuit resulting in the same IP (exit node).

Is there a way to tell it to generate new one?

Thanks

**Comment by ProjectDarkTiger** [comment]
I want to know too!

**Comment by XSPGMike** [comment]
I have the same question

**Comment by nguyenhn1299** [comment]
You can check the Dockerfile that the controlport is 9051. So just map it outside to use.

`sudo docker run -it -p 8118:8118 -p 9050:9050 -p 9051:9051 -e PASSWORD=yourpass dperson/torproxy
`
```
from stem import Signal
from stem.control import Controller

with Controller.from_port(port=9051) as controller:
	controller.authenticate(password="yourpass")
	controller.signal(Signal.NEWNYM)
```

**Comment by des1redState** [comment]
Yes. See comment in thread: https://github.com/dperson/torproxy/issues/80

**Comment by henriqueutsch** [comment]
Hy guys
My question is 9051 port is not open when I run nmap command
`    image: dperson/torproxy
    ports:
      - 8181:8181
      - 9050:9050
      - 9051:9051
    environment:`
